package/gluon-core/luasrc/lib/gluon/upgrade/820-dns-config

```diff
@@ -0,0 +1,37 @@
+#!/usr/bin/lua
+local site = require 'gluon.site_config'
+local uci = require('luci.model.uci').cursor()
+
+if site.dns.servers
+then
+	dnsmasq=uci:get_first("dhcp", "dnsmasq")
+	uci:set('dhcp', dnsmasq, 'server', site.dns.servers)
+	uci:set('dhcp', dnsmasq, 'localise_queries', '1')
+	uci:set('dhcp', dnsmasq, 'localservice', '0')
+
+	if site.dns.cacheentries
+	then
+		uci:set('dhcp', dnsmasq, 'cachesize', site.dns.cacheentries)
+	end
+
+	if site.next_node.name
+	then
+		if site.next_node.ip4
+		then
+			uci:section('dhcp','domain','nextnode4',{
+				name=site.next_node.name,
+				ip=site.next_node.ip4,
+			}
+			)
+		end
+		if site.next_node.ip6
+		then
+			uci:section('dhcp','domain','nextnode6',{
+				name=site.next_node.name,
+				ip=site.next_node.ip6,
+			}
+			)
+		end
+	end
+	uci:save('dhcp')
+end
```

Comment:
I think most of these options don't need to depend on *site.dns.servers*. We can just unconditionally set *localise_queries* and *localservice*, and set each of the other options/sections when it is given in *site.conf*.

We would also need to disable *peerdns* for the client interface when *site.dns.servers* is given, to avoid the node picking up the next-node address as upstream DNS.

Please also add code to remove/reset all these options when the corresponding values don't exist in *site.conf* (anymore).